### ISSUE TITLE
Possibility to add optional labels

### ISSUE THREAD

**Issue by andrikaro** [burningalchemist/sql_exporter#317]
**Is your feature request related to a problem? Please describe.**

While writing and testing collectors I discovered that when query returns null for a field that is defined under key_labels the query fails and no value is pulled even if it exists. 

**Describe the solution you'd like**

As all of the labels are not strictly required I wonder if it could default to just omitting the label. 

Or perhaps there is a way to configure a label as optional.

**Describe alternatives you've considered**

Currently I found a workaround like this:
```yaml
key_labels:
  - user
query: |
  SELECT COALESCE(user, '~unknown') FROM table;
```

**Additional context**

Of course this issue will be indirectly remedied by [this](https://github.com/burningalchemist/sql_exporter/issues/256).

**Comment by burningalchemist** [comment]
Hey @andrikaro, thank you for your suggestion and the workaround posted! 👍

It is tricky to make labels optional as Prometheus devs encourage consistency (which is good anyway). There is a thread in Google Groups specifically about adding/removing labels dynamically for existing metric families. Overall, there's no way to achieve this and if there's one it's discouraged.

What we actually can do is to handle null values properly, and return empty strings for labels which aren't populated. I believe this should solve your issue and still keep labels consistent.

I plan to release `v0.12.2` this week, and then switch to `v0.13` to introduce error metrics. I'll try to test it first, and then probably it becomes a part of the next minor release. 👍

**Comment by burningalchemist** [comment]
Hey @andrikaro, it's available in `v0.12.3`, thanks to some regressions. 😅 Please, give it a shot. :)
